freetextresponse/freetextresponse.py

```diff
@@ -0,0 +1,437 @@
+        ):
+            return 'correct'
+        else:
+            return 'incorrect'
+
+    def word_count_valid(self):
+        '''
+        Returns a boolean value indicating whether the current
+        word count of the user's answer is valid
+        '''
+        word_count = len(self.student_answer.split())
+        return (
+            word_count <= self.max_word_count and
+            word_count >= self.min_word_count
+        )
+
+    def phrase_present_in_answer(self, phrase_list):
+        '''
+        Helper method that returns a boolean value indicating
+        whether at least one member of the provided phrase list
+        is present in the student's answer
+        '''
+        if re.search(
+                '|'.join(
+                    [phrase.strip() for phrase in phrase_list.split(',')]
+                ),
+                self.student_answer,
+                re.IGNORECASE
+        ):
+            return True
+        return False
+
+    def get_problem_progress(self):
+        '''
+        Returns a statement of progress for the XBlock, which depends
+        on the user's current score
+        '''
+        if self.score == 0.0:
+            return (
+                str(self.weight)
+                + ' point'
+                + ('s' if (self.weight > 1) else '')
+                + ' possible'
+            )
+        else:
+            return (
+                str(self.score).replace('.0', '')
+                + '/' + str(self.weight)
+                + ' point'
+                + ('s' if (self.weight > 1) else '')
+            )
+
+    def compute_score(self):
+        '''
+        Computes and publishes the user's core for the XBlock
+        based on their answer
+        '''
+        credit = self.determine_credit()
+        if credit == Credit.full:
+            self.score = self.weight
+        elif credit == Credit.half:
+            self.score = float(self.weight)/2
+        else:
+            self.score = 0.0
+        self.runtime.publish(
+            self, "grade",
+            {
+                'value': self.score,
+                'max_value': self.weight
+            }
+        )
+
+    def determine_credit(self):
+        '''
+        Helper Method that determines the level of credit that
+        the user should earn based on their answer
+        '''
+        if self.student_answer == '' or not self.word_count_valid():
+            return Credit.zero
+        if self.phrase_present_in_answer(self.fullcredit_keyphrases):
+            return Credit.full
+        elif self.phrase_present_in_answer(self.halfcredit_keyphrases):
+            return Credit.half
+        else:
+            return Credit.zero
+
+    def get_used_attempts_feedback(self):
+        '''
+        Returns the text with feedback to the user about the number of attempts
+        they have used if applicable
+        '''
+        if self.max_attempts == 0:
+            return ''
+        return (
+            'You have used '
```

Comment:
Not only should this be `i18n`ed, it should use `ngettext` to handle pluralization.